packages/experimental/src/useController.ts

```diff
@@ -0,0 +1,70 @@
+import { EndpointInterface } from '@rest-hooks/endpoint';
+import { DispatchContext } from '@rest-hooks/core';
+import { useContext, useCallback } from 'react';
+import { actionTypes } from '@rest-hooks/core';
+
+import createFetch from './createFetch';
+import { UpdateFunction } from './types';
+
+/**
+ * Imperative control of Rest Hooks store
+ * @see https://resthooks.io/docs/api/useController
+ */
+export default function useController(throttle = false): Controller {
+  const dispatch = useContext(DispatchContext);
+
+  // TODO: elevate into CacheProvider and simply useContext
+  const fetch = useCallback(
+    <E extends EndpointInterface>(
+      endpoint: E,
+      ...args: readonly [...Parameters<E>]
+    ) => {
+      const action = createFetch(endpoint, {
+        args,
+        throttle,
+      });
+      dispatch(action);
+
+      return action.meta.promise as any;
+    },
+    [dispatch, throttle],
+  );
+
+  const invalidate = useCallback(
+    <E extends EndpointInterface>(
+      endpoint: E,
+      ...args: readonly [...Parameters<E>] | readonly [null]
+    ) =>
+      args?.[0] !== null
```

Comment:
```suggestion
      args[0] !== null
```
it's impossible that `args` is not an array

Reply:
Are there no out of bounds checks in Javascript?

Reply:
Of course not, arrays are just objects. Either way, `?.` does nothing different unless the LHS is null or undefined.